audbcards/core/dataset.py

```diff
@@ -16,6 +18,32 @@
 from audbcards.core.utils import limit_presented_samples
 
 
+def datasetcache(func):
+    r"""Decorator to manage dataset instance cache."""
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        """Inner Function containing logic."""
+        cache_root =  audeer.mkdir(audeer.safe_path(kwargs.get('cache_root')))
+        dbname, version = args
+        cache_filename = audeer.path(
```

Comment:
The path to the cache file is currently defined at two places, which is not a good idea.
So maybe create a method for defining the cache path.

Reply:
This should be fixed.